src/toil/batchSystems/slurm.py

```diff
@@ -467,6 +489,25 @@ def parse_elapsed(self, elapsed):
                 pass  # slurm may return INVALID instead of a time
             return total_seconds
 
+    def __init__(self, config, maxCores, maxMemory, maxDisk):
+        super().__init__(config, maxCores, maxMemory, maxDisk)
+        self._get_partition_info()
+
+    def _get_partition_info(self):
+        sinfo_command = ["sinfo",
+                         "-a",
+                         "-o",
+                         "%P %G %l %p %c %m"]
+
+        sinfo = call_command(sinfo_command)
+
+        self.partitions = []
+        PartitionInfo = namedtuple("PartitionInfo", ["partition_name", "gres", "time", "priority", "cpus", "memory"])
```

Comment:
Instead of minting the type at runtime it probably makes more sense to use `typing.NamedTuple` so this can be a `class`: https://docs.python.org/3/library/typing.html#typing.NamedTuple